### PR TITLE
feat: Configure Mobile IAP cronjob to create iOS products on AppStore

### PR DESCRIPTION
**Pull request by moeez96**
# ⛔️ MAIN BRANCH WARNING! 2U EMPLOYEES must make branches against the 2u/main BRANCH

- [ ] I have checked the branch to which I would like to merge.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py [hunk 1]
```diff
@@ -62,76 +77,133 @@ def handle(self, *args, **options):
...
+        self._send_email_about_expired_courses(expired_courses_keys, all_course_runs_processed,
+                                               failed_course_runs, new_seats_created, failed_ios_products)
 
-    def _get_parent_products_to_create_mobile_skus_for(self, courses):
+    def _get_course_run_keys_for_parent_course(self, course, default_site):
+        """
+        Get parent course key from discovery for the current course run.
+        Get all course run keys for parent course from discovery. Then filter those
+        courses/course runs on Ecommerce using Course.verification_deadline and
+        Product.expires to determine products to create course runs for.
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: *mobile products

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py [hunk 2]
```diff
@@ -62,76 +77,133 @@ def handle(self, *args, **options):
...
-            expired_course.publish_to_lms()
             batch_counter += 1
             if batch_counter >= batch_size:
                 time.sleep(sleep_time)
                 batch_counter = 0
+        self._send_email_about_expired_courses(expired_courses_keys, all_course_runs_processed,
+                                               failed_course_runs, new_seats_created, failed_ios_products)
 
-    def _get_parent_products_to_create_mobile_skus_for(self, courses):
+    def _get_course_run_keys_for_parent_course(self, course, default_site):
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: a simpler name can be _get_related_course_runs

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py [hunk 3]
```diff
@@ -62,76 +77,133 @@ def handle(self, *args, **options):
...
+                    continue
+
+                if not mobile_products:
+                    failed_course_runs.append(course_run.id)
+                else:
+                    android_sku = list(filter(lambda sku: 'android' in sku.partner_sku, mobile_products))[0].partner_sku
+                    ios_sku = list(filter(lambda sku: 'ios' in sku.partner_sku, mobile_products))[0].partner_sku
+                    new_seats_created.append("{},{},{}".format(ios_sku, android_sku, course_run.id))
+
+                    error_message = self._create_ios_product(mobile_products, course_run, default_site)
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: can we send just ios_product since we are already a filter above?

**Review by jawad-khan — APPROVED** [review]

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py (hunk 1 above)
Comment (moeez96) [reply ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: Done

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py (hunk 3 above)
Comment (moeez96) [reply ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: Done

File: ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py (hunk 2 above)
Comment (moeez96) [reply ecommerce/extensions/iap/management/commands/batch_update_mobile_seats.py]: Renamed as _get_related_course_run_keys

**Review by zubair-ce07 — APPROVED** [review]

**Pull request merged**